UnwrapBlog.md

```diff
@@ -0,0 +1,92 @@
+Advanced TypeScript type FTW!
+
+In Pulumi we have a data model that allows people to express complex data that may *eventually* be available. Traditional JS programming might expose that as a Promise<T>, but we've taken that one step further by introducing a type we call:
+
+```ts
+type Input<T> = T | Promise<T> | Output<T>;
+
+// Like a Promise<T>, but also keeps track of Dependency information so we can
+// figure out what the upstream/downstream impact is of any value change.
+interface Output<T> { /* ... */ }
+```
+
+This complex type can often show up in forms like:
```

Comment:
Maybe

> By convention, all resource input properties are `Input<T>`s, and al output properties are `Output<T>`s. These complex types can often show up in forms like:

?

Reply:
I didn't want to bring up 'Resources', because that's a whole new concept in itself.

Reply:
That's why i just leave it as Inputs, and Outputs, and just briefly describe why an 'Output' would even exist.

Reply:
I like something like what @joeduffy suggests - maybe just "all inputs to Pulumi objects"?